test/sentiment.js

```diff
@@ -0,0 +1,169 @@
+        await sentimentAnalysis.getReputation(accounts[1],
+            {
+                from: ownerAddress
+            }
+        )
+        .then(function(result) {
+            assert.equal(result[0], ""); // reputation
+            assert.equal(result[1], 0); // correct
+            assert.equal(result[2], 0); // incorrect
+            assert.equal(result[3], ""); // lastUpdatedDate
+            assert.equal(result[4], ""); // lastFormulaApplied
+        });
+    });
+
+    it('should not be able to update reputations if not owner', async function() {
+        await assertFail(async function() {
+            await sentimentAnalysis.updateReputation(
+                "0.525", 1, 0, todayISO, "0.5 + (0.5*0.05)", accounts[1],
+                {
+                    from: accounts[1]
+                }
+            )
+        });
+    });
+
+    it('should be able to update reputations if owner', async function() {
+        await sentimentAnalysis.updateReputation(
+            "0.525", 1, 0, todayISO, "0.5 + (0.5*0.05)", accounts[1],
+            {
+                from: ownerAddress
+            }
+        )
+        .then(result => {
+            eventsUtil.eventValidator(
+                result, 
+                {
+                    name: "ReputationUpdated",
+                    args: {
+                        reputation: "0.525",
+                        correct: 1,
+                        incorrect: 0,
+                        lastUpdateDate: todayISO,
+                        lastFormulaApplied: "0.5 + (0.5*0.05)",
+                        user: accounts[1]
+                    }
+                }
+            );
+        });
+    });
+    
+    it('should be able to retrieve reputation for user if owner', async function() {
+        await sentimentAnalysis.getReputation(accounts[1],
+            {
+                from: ownerAddress
+            }
+        )
+        .then(result => {
+            assert.equal(result[0], "0.525"); // reputation
+            assert.equal(result[1], 1); // correct
+            assert.equal(result[2], 0); // incorrect
+            assert.equal(result[3], todayISO); // lastUpdatedDate
+            assert.equal(result[4], "0.5 + (0.5*0.05)"); // lastFormulaApplied
+        });
+    });
+
+    it('should be able to retrieve reputation about himself if ordinary user', async function() {
+        await sentimentAnalysis.getReputation(accounts[1],
+            {
+                from: accounts[1]
+            }
+        )
+        .then(result => {
+            assert.equal(result[0], "0.525"); // reputation
+            assert.equal(result[1], 1); // correct
+            assert.equal(result[2], 0); // incorrect
+            assert.equal(result[3], todayISO); // lastUpdatedDate
+            assert.equal(result[4], "0.5 + (0.5*0.05)"); // lastFormulaApplied
+        });
+    });
+
+    it('should not be able to retrieve reputation about somebody else if ordinary user', async function() {
```

Comment:
As above, anyone should be able to read all reputations, to verify there's no "cheating"